pkg/manifests/template/cache.go

```diff
@@ -1,27 +1,36 @@
 package template
 
 import (
+	"sync"
+
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 var (
 	namespacedCache = &gvkCache{cache: map[schema.GroupVersionKind]bool{}}
+	mapMutex        = sync.RWMutex{}
```

Comment:
there's also a concurrent map we can use that might do this more efficiently

Reply:
I think it's used in the other caches, i didn't even think about the need for it here